Error: unsupported instruction `sysenter'

**Describe the bug**
when I try to compile dynamorio it aborts with an error:

```
# make
-- Version number: 7.91.18289
-- WARNING: vera++ not found: disabling code style checks
-- Found Qt 5: DrGUI will be built
[...]
prg/tmp/dynamorio/core/arch/x86/x86.asm: Assembler messages:
/prg/tmp/dynamorio/core/arch/x86/x86.asm:805: Error: unsupported instruction `sysenter'
make[2]: *** [core/CMakeFiles/dynamorio_static.dir/build.make:822: core/CMakeFiles/dynamorio_static.dir/arch/x86/x86.asm.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:143: core/CMakeFiles/dynamorio_static.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
```
I thought this is make due a change in the code and reverted to an old version of dynamorio that previously compiled fine for me (dynamorio-cronbuild-7.90.18005 from April 2019) - but same issue there. 

I also tried gcc-7 and gcc-9, same outcome.

My guess its about the kernel/libc/headers? I am running Linux 5.4.0-amd64 and using libc 2.29

**To Reproduce**
```
git clone https://github.com/DynamoRIO/dynamorio/
cd dynamorio
mkdir build
cd build
cmake ..
make
```

**Versions**
github checkout, but same for older versions e.g. 7.0 and 7.1 releases.

--- a/core/arch/x86/x86.asm
+++ b/core/arch/x86/x86.asm
@@ -796,11 +796,7 @@ GLOBAL_LABEL(global_do_syscall_sygate_int:)
  */
         DECLARE_FUNC(global_do_syscall_sysenter)
 GLOBAL_LABEL(global_do_syscall_sysenter:)
-#if defined(X64)
-        syscall  /* FIXME ml64 won't take "sysenter" so half-fixing now */
-#else
-        sysenter
-#endif
+        RAW(0f) RAW(34) /* sysenter */
 #ifdef DEBUG
         /* We'll never ever reach here, sysenter won't/can't return to this
          * address since it doesn't know it, but we'll put in a jmp to

--- a/core/arch/x86/x86.asm
+++ b/core/arch/x86/x86.asm
@@ -796,7 +796,7 @@ GLOBAL_LABEL(global_do_syscall_sygate_int:)
  */
         DECLARE_FUNC(global_do_syscall_sysenter)
 GLOBAL_LABEL(global_do_syscall_sysenter:)
-#if defined(X64) && defined(WINDOWS)
+#if defined(X64)
         syscall  /* FIXME ml64 won't take "sysenter" so half-fixing now */
 #else
         sysenter

